Métodos para alterar senha e cadastrar dados pessoais do empregado

gerorsch opened the pull request:
Adiciona métodos para cadastrar dados pessoais do empregado e alterar sua senha.

* A função de {@code cadastrarEmpregado} cria e adiciona um objeto do tipo {@code empregado} com senha no formato hash, em especial com o algoritmo SHA-256, dessa forma, armazena apenas o digest da senha com a função hash. Há também a verificação se o cpf é válido, com a validação básica de um número cpf. O objeto criado por essa função é  armazenado no objeto {@code repoEmpregado}, que se trata do repositório/banco de dados de empregados.

* Método que altera os dados cadastrados de um empregado por meio da substituição do objeto {@code Empregado} antigo por um novo objeto do tipo {@code Empregado}.

gerorsch committed:
feat(beans): Construtor Empregado Clone

gerorsch committed:
feat: Método alterarSenhaCliente()

*Novo método para alterar uma senha específica de um empregado
*Cria uma exceção caso use um uidEmpregado não existente

gerorsch committed:
Método alterarDadosPessoais()

Método que altera os dados cadastrados de um empregado por meio da substituição do objeto {@code Empregado} antigo por um novo objeto do tipo {@code Empregado}.

SteffanoP commented on src/negocio/ControladorEmpregado.java:
```diff
@@ -15,6 +15,55 @@ public ControladorEmpregado() {
+
+        try {
+            MessageDigest algoritmoEncrypt = MessageDigest.getInstance("SHA-256");
+            byte[] senhaDisgest = algoritmoEncrypt.digest(senhaCliente.getBytes(StandardCharsets.UTF_8));
```
```suggestion
            byte[] senhaDisgest = algoritmoEncrypt.digest(senhaEmpregado.getBytes(StandardCharsets.UTF_8));
```

SteffanoP commented on src/negocio/ControladorEmpregado.java:
```diff
@@ -31,8 +80,34 @@ public String informacoesPessoais(Empregado empregado) {
+     */
+    public void alterarDadosPessoais(long uidEmpregado, Empregado empregadoDadosNovo) throws EmpregadoInexistenteException {
+        //TODO: Revisar regras de negócio e se aplicam neste espaço
+        for (Emopregado empregado : this.repoEmpregado.listar()) {
```
```suggestion
        for (Empregado empregado : this.repoEmpregado.listar()) {
```

gerorsch committed:
Update src/negocio/ControladorEmpregado.java

Co-authored-by: Steffano Pereira <steffanoxpereira@gmail.com>

gerorsch committed:
Fix: src/negocio/ControladorEmpregado.java

Co-authored-by: Steffano Pereira <steffanoxpereira@gmail.com>

gerorsch committed:
Cria Exceptions para Empregado

Adiciona exceptions EmpregadoDuplicadoException e EmpregadoCPFInvalidoException.

gerorsch committed:
Merge branch 'dev-metodos-controlados-empregado' of https://github.com/SteffanoP/projetoIP2 into dev-metodos-controlados-empregado

SteffanoP committed:
fix(Exceptions): Import de exceptions
